### ISSUE TITLE
Using same routematcher parameter, cy.wait doesn't wait until the latest alias intercepts request

### ISSUE THREAD

**Issue by quad5** [cypress-io/cypress#14790]
### Current behavior

When I have multiple aliases using the same route matcher parameter, cy.wait doesn't wait for the alias I want. Please see pictures.

<img width="916" alt="Untitled11" src="https://user-images.githubusercontent.com/2043287/106062471-ae9b0080-60ab-11eb-85d0-53df33136d71.png">

<img width="913" alt="Untitled10" src="https://user-images.githubusercontent.com/2043287/106062463-ac38a680-60ab-11eb-8122-dcb868ebbde1.png">

Eventually, I get this error message (below image is from different run than the one above):
<img width="849" alt="Untitled12" src="https://user-images.githubusercontent.com/2043287/106667208-7db04500-655d-11eb-97fa-e0660f66d7f4.png">

### Desired behavior

I would expect it to wait for latest alias.

### Test code to reproduce

```js
it('test', function() {
    cy.intercept({routeMatcherObject}, (req) => {
        //do something
    }).as('alias')
    cy.visit('some url')
    cy.wait('@alias')
    cy.intercept({routeMatcherObject}, (req) => {
        // do something
    }).as('alias2')
    cy.get('some locator').click()
    cy.wait('@alias2')
})
```

### Versions

Cypress: 6.3.0
Chrome: 88.0.4324.96
Mac: 11.0.1

**Comment by DRiewaldt** [comment]
I just want to confirm this is an issue, and I've seen it happen. If the first routeMatcher has the same as the second routeMatcher in the same it block, the second aliased wait won't work as expected.

For me it threw an "cy.its() errored because the propert: 'response' does not exist on your subject".

Image from my Cypress test. 

![image](https://user-images.githubusercontent.com/62670621/106526841-85131800-64a3-11eb-8a0a-90427c35fb6a.png)

Example code:
```
 it(`should verify stuff`, () => {
            cy.intercept('POST', '/api_stuff/stuffandthings').as('exampleCreation');
            cy.get('.confirm').should('be.visible').click();
            cy.wait('@exampleCreation').its('response.statusCode').should('eq', 200);

            cy.get('.aClass').should('be.visible').invoke('text').then(data => {
                //some code
            });
            cy.get('.aClass2').should('be.visible').invoke('text').then(data => {
                // some code
            });
            cy.get('.content > :nth-child(2)').should('be.visible').invoke('text').then(data => {
                // some code
            });
            cy.get('.cancel').should('be.visible').invoke('text').then(data => {
                // some code
            });
            
            cy.intercept('POST', '/api_stuff/stuffandthings').as('confirmCreation') // this won't bind to this alias listed, it will instead bind to the first alias exampleCreation
            cy.get('.confirm').should('be.visible').click();
            cy.wait('@confirmCreation').its('response.statusCode').should('eq', 200); // this will be undefined, and claim there is no response
        });
```

**Comment by renanbandeira** [comment]
I'm facing the same issue as you, @DRiewaldt 
It happens because the interception object isn't the same as the documentation:
![image](https://user-images.githubusercontent.com/1872284/109156266-66750980-774f-11eb-8d7e-c6c3fc1ee65e.png)
The image above shows how is the object coming from wait request

**Comment by flotwig** [comment]
@quad5 I think this is a duplicate of #9302, which will be fixed in Cypress 7.0. Request handler ordering will be reversed and this will allow you to override request handlers as expected.

@DRiewaldt @renanbandeira This is a known issue that is also fixed in Cypress 7.0 with #15528 

Closing as duplicate.
